front/lib/api/assistant/conversation.ts

```diff
@@ -1336,6 +1352,13 @@ export async function* editUserMessage(
         row: AgentMessage;
         m: AgentMessageWithRankType;
       }[];
+
```

Comment:
I revised my stance on editUserMessage => currently, it does not delete former agent messages if mention is changed so the convo should not loosen its permission.

We'd have to be careful in the future when deletion comes and depending on how we implement it, e.g. if we still keep the agent messages versions when deleted just as we keep the various messages versions, then we'd actually keep the "purely additive" stance and *not* remove permission restriction

Thinking about it more, it seems better: simpler code, clearer additive model, and user downside very limited (only case is: now that the message is deleted user should be able to access the convo but can't, seems very edge-casey and easily overcomable)

Reply:
Will edit the PR description if there's agreement

Reply:
Let's leave a comment here?